addons/l10n_vn_edi/models/account_move.py

```diff
@@ -0,0 +1,719 @@
+            'discount': 3,
+        }
+        for line in self.invoice_line_ids.filtered(lambda ln: ln.display_type != 'line_section'):
+            # For credit notes amount, we send negative values (reduces the amount of the original invoice)
+            sign = 1 if self.move_type == 'out_invoice' else -1
+            item_information = {
+                'itemCode': line.product_id.code,
+                'itemName': line.product_id.name,
+                'unitName': line.product_uom_id.name,
+                'unitPrice': line.price_unit * sign,
+                'quantity': line.quantity,
+                # This amount should be without discount applied.
+                'itemTotalAmountWithoutTax': line.currency_id.round(line.price_unit * line.quantity) * sign,
+                # In Vietnam a line will always have only one tax.
+                # Values are either: -2 (no tax), -1 (not declaring/paying taxes), 0,5,8,10 (the tax %)
+                # Most use cases will be -2 or a tax percentage, so we limit the support to these.
+                'taxPercentage': line.tax_ids and line.tax_ids[0].amount or -2,
+                'taxAmount': (line.price_total - line.price_subtotal) * sign,
+                'discount': line.discount,
+                'itemTotalAmountAfterDiscount': line.price_subtotal * sign,
+                'itemTotalAmountWithTax': line.price_total * sign,
+            }
+            if line.display_type in code_map:
+                item_information['selection'] = code_map[line.display_type]
+            if line.display_type == 'discount':
+                item_information['isIncreaseItem'] = False
+            if self.move_type == 'out_refund':
+                item_information.update({
+                    'adjustmentTaxAmount': item_information['taxAmount'],
+                    'isIncreaseItem': False,
+                })
+            items_information.append(item_information)
+
+        return items_information
+
+    def _l10n_vn_edi_get_tax_breakdowns(self):
+        """ Create and return the tax breakdown of the current invoice. """
+        self.ensure_one()
+
+        def grouping_key_generator(base_line, tax_values):
+            # Requirement is to generate a tax breakdown per taxPercentage
+            return {'tax_percentage': tax_values['tax_repartition_line'].tax_id.amount or -2}
+
+        tax_breakdowns = []
+
+        tax_details_grouped = self._prepare_invoice_aggregated_taxes(grouping_key_generator=grouping_key_generator)
+        for tax_percentage, tax_percentage_values in tax_details_grouped['tax_details'].items():
+            tax_breakdowns.append({
+                'taxPercentage': tax_percentage['tax_percentage'],
+                'taxableAmount': tax_percentage_values['base_amount'],
+                'taxAmount': tax_percentage_values['tax_amount'],
+                'taxableAmountPos': self.move_type == 'out_invoice',  # For adjustment invoice, the amount should be considered as negative.
```

Comment:
base_amount / tax_amount are expressed in company's currency. Is it really what you want?

Reply:
Hmm probably no, nice catch. It should be the invoice currency like the rest of it.